traefik-golang/backend/Dockerfile

```
@@ -1,9 +1,27 @@
-FROM golang:1.13 AS build
+# syntax=docker/dockerfile:1.4
+
+FROM --platform=$BUILDPLATFORM golang:1.18 AS build
 
 WORKDIR /compose/hello-docker
 COPY main.go main.go
 RUN CGO_ENABLED=0 go build -o backend main.go
 
+FROM build as dev-envs
+
+RUN <<EOF
+apt update
+apt install git
```

```suggestion
apt-get update
apt-get install -y --no-install-recommends git
```

* `apt` likes to cry that it doesn't have a stable CLI so isn't meant for scripts
* `git` has a ton of recommended packages so needs the extra flags (works for this image because `golang` already includes Git - try out `docker run --rm node:17.0.1-bullseye-slim sh -c 'apt-get update && apt-get install git'` to see failure you'd get otherwise)